### ISSUE TITLE
[CLI] Bidding interface wantCollateral required when bidding by-price

### ISSUE THREAD

**Issue by arirubinstein** [Agoric/Agoric#7268]
## Describe the bug
The bidding interface is confusing currently, when specifying a price, it also requires the wantCollateral.  If this information can be implied by the price argument and `giveCurrency` param, requiring the user to provide this is not ergonomic.

## To Reproduce
1. `agops inter bid by-price --price 0.80 --giveCurrency 10`
2.  see error about required `wantCollateral`

## Expected behavior
A bidding user should be able to specify the minimum input required to make a bid instead of having to calculate the wanted collateral

**Comment by dckc** [comment]
I tried tweaking the CLI this way, but then I got: `Error: (an object) - Must have missing properties ["want"]`

This seems to be a contract issue as well as CLI:

https://github.com/Agoric/agoric-sdk/blob/b834e89d39edf2af1a0a730f33b68494fe0c9ee2/packages/inter-protocol/src/auction/auctionBook.js#L71

cc @Chris-Hibbert @turadg

**Comment by Chris-Hibbert** [comment]
I'm not convinced that the bidSpec's want (though admittedly mis-named) is extraneous. There are two ways of specifying a bid: by price and by discount
 * I'm willing to spend 37 IST per ATOM, and I want 100. I might get only 50, and they might only cost me 35 IST each. I don't have to give 3700 IST in order to make this bid.
 * I'd like to buy ATOM at a 80% of the oracle price, and I'd like up to 1000. The amount of IST I provide is unrelated to how many I'd like to acquire.

**Comment by dckc** [comment]
fixed in #7256:

```
$ agops inter bid by-price --price 0.80 --give 10IST --from gov1
2023-04-05T18:11:44.932548675Z tx not in block 5617 retrying...
```
